examples/vc0706_snapshot_filesystem.py

```diff
@@ -1,92 +1,56 @@
+elif TARGET == "PC":
+    # USB UART
+    # Be sure to modify this to the correct device! (Windows COM, etc)
+    import serial
+
+    UART = serial.Serial("/dev/ttyUSB0", baudrate=115200, timeout=0.25)
+    IMAGE_FILE = "image.jpg"  # USB storage.
+elif TARGET == "PI":
+    # Pi Hardware UART
+    import serial
+
+    UART = serial.Serial("/dev/ttyS0", baudrate=115200, timeout=0.25)
+    IMAGE_FILE = "/home/pi/image.jpg"  # Pi storage
 
 # Setup VC0706 camera
-vc0706 = adafruit_vc0706.VC0706(uart)
+vc0706 = adafruit_vc0706.VC0706Camera(UART)
 
 # Print the version string from the camera.
 print("VC0706 version:")
 print(vc0706.version)
 
-# Set the image size.
-vc0706.image_size = adafruit_vc0706.IMAGE_SIZE_640x480
-# Or set IMAGE_SIZE_320x240 or IMAGE_SIZE_160x120
-
-# Note you can also read the property and compare against those values to
-# see the current size:
-size = vc0706.image_size
-if size == adafruit_vc0706.IMAGE_SIZE_640x480:
-    print("Using 640x480 size image.")
-elif size == adafruit_vc0706.IMAGE_SIZE_320x240:
-    print("Using 320x240 size image.")
-elif size == adafruit_vc0706.IMAGE_SIZE_160x120:
-    print("Using 160x120 size image.")
-
 # Take a picture.
 print("Taking a picture in 3 seconds...")
 time.sleep(3)
 print("SNAP!")
-if not vc0706.take_picture():
-    raise RuntimeError("Failed to take picture!")
-
-# Print size of picture in bytes.
-frame_length = vc0706.frame_length
-print("Picture size (bytes): {}".format(frame_length))
-
-# Open a file for writing (overwriting it if necessary).
-# This will write 50 bytes at a time using a small buffer.
-# You MUST keep the buffer size under 100!
-print("Writing image: {}".format(IMAGE_FILE), end="", flush=True)
+print("Saving image. This may take some time.")
 stamp = time.monotonic()
-# Pylint doesn't like the wcount variable being lowercase, but uppercase makes less sense
-# pylint: disable=invalid-name
-with open(IMAGE_FILE, "wb") as outfile:
-    wcount = 0
-    while frame_length > 0:
-        t = time.monotonic()
-        # Compute how much data is left to read as the lesser of remaining bytes
-        # or the copy buffer size (32 bytes at a time).  Buffer size MUST be
-        # a multiple of 4 and under 100.  Stick with 32!
-        to_read = min(frame_length, 32)
-        copy_buffer = bytearray(to_read)
-        # Read picture data into the copy buffer.
-        if vc0706.read_picture_into(copy_buffer) == 0:
-            raise RuntimeError("Failed to read picture frame data!")
-        # Write the data to SD card file and decrement remaining bytes.
-        outfile.write(copy_buffer)
-        frame_length -= 32
-        # Print a dot every 2k bytes to show progress.
-        wcount += 1
-        if wcount >= 64:
-            print(".", end="", flush=True)
-            wcount = 0
-print()
-# pylint: enable=invalid-name
+imgfile = open(IMAGE_FILE, "wb")
+vc0706.take_picture(imgfile, 640, 480)  # You can also use 320 x 240 or 160 x 120
+imgfile.close()
```

Comment:
Use a with block here and in the other example. It's better because the file will be closed even if an exception happens.

```suggestion
with open(IMAGE_FILE, "wb") as imgfile:
    vc0706.take_picture(imgfile, 640, 480)  # You can also use 320 x 240 or 160 x 120
```

Reply:
Looks nicer, too. Done.